### PR TITLE
Scout changes

### PR DESCRIPTION
**Pull request by AhmedHamed3699**

### CODE REVIEW

File: server/database/db.ts
Comment (amir-kedis) [inline server/database/db.ts:1]: Can't this module get refactored into a map / dict and extracted in a more obvious way (to reduce the amount of nested long if else chains)

File: server/database/scheduler.ts
Comment (amir-kedis) [inline server/database/scheduler.ts:1]: This file probably should be deleted

**Review by amir-kedis — COMMENTED** [review]
Great work ya hamed look at comments please

File: server/database/db.ts
Comment (AhmedHamed3699) [reply server/database/db.ts:1]: I'll try, hoping one day TypeScript just go right to hell

Comment (amir-kedis) [reply server/database/db.ts:1]: 3azmaa ya Sadiki 🔥🔥

File: server/utils/getScoutsByName.ts
Comment (amir-kedis) [inline server/utils/getScoutsByName.ts:1]: This is a file where there is the conflict @akramhany  adjusted this part in his pr #80 to be able to get scouts in just your unit of control (EX. unit view scouts only in his unit,  regular in his sector, etc. the problem now with his implementation is that hi uses old scout schema so when you resolve the conflict probably keep his logic while maintaining your new schema

**Review by amir-kedis — COMMENTED** [review]
:see_no_evil: see the conflict I left a comment in the file :smile:

File: server/utils/getScoutsByName.ts
Comment (AhmedHamed3699) [reply server/utils/getScoutsByName.ts:1]: I think it is resolved now :)

**Review by amir-kedis — APPROVED** [review]
Great work ya Hamed :fire: 
to end this branch I will merge it into a new global feature branch to not break the production but you can consider the task done :smile:

**Pull request merged**